Source/WebCore/svg/SVGPathElement.cpp

```diff
@@ -217,4 +219,14 @@ RenderPtr<RenderElement> SVGPathElement::createElementRenderer(RenderStyle&& sty
     return createRenderer<LegacyRenderSVGPath>(*this, WTFMove(style));
 }
 
+Path SVGPathElement::path() const
+{
+    if (auto* renderer = this->renderer()) {
+        if (auto* basicShapePath = renderer->style().d())
+            return basicShapePath->path({ });
+    }
+
+    return m_pathSegList->currentPath();
+}
```

Comment:
Can you explain what is this function returning? Are we allowing the `m_pathSegList` to be out of sync from the style property? What should JS return in case of SMIL and CSS animations? What is the expected value of
```
pathSegList
animatedPathSegList
getAttribute('d')
```

Reply:
I made a [little test](https://codepen.io/graouts/pen/BaEVOBe) for this, dumping the various value half-way through an animation performed by either an SVG `<animate>` element or a CSS Animation from `M 0 0 H 100 V 100 H 0 Z` to `M 0 0 H 200 V 200 H 0 Z` (values rounded for legibility):

```
Path animated with SVG
    pathSegList = M 0 0 H 100 V 100 H 0 z
    animatedPathSegList = M 0 0 H 150 V 150 H 0 z
    getAttribute("d") = M 0 0 H 100 V 100 H 0 Z
    computed style = path("M 0 0 H 150 V 150 H 0 Z")

Path animated with CSS
    pathSegList = M 0 0 H 100 V 100 H 0 z
    animatedPathSegList = M 0 0 H 100 V 100 H 0 z
    getAttribute("d") = M 0 0 H 100 V 100 H 0 Z
    computed style = path("M 0 0 H 150 V 150 H 0 Z")
```

The `getAttribute("d")` and computed style for the `d` property match what I see in Chrome Canary and Firefox Nightly. Both of those browsers having dropped support for `pathSegList` and `animatedPathSegList` I cannot compare those values.

Reply:
Note that I did not introduce this function, I just moved its implementation in the `.cpp` file since it now had more code.